### ISSUE TITLE
carta command line argument does not understand absolute path?

### ISSUE THREAD

**Issue by teuben** [CARTAvis/carta#31]
the command
      carta /tmp/test6503.fits
fails, but relative and local path references seem to work.

**Comment by kswang1029** [comment]
@ajm-asiaa could you check this?

**Comment by ajm-ska** [comment]
This is quite strange. I confirmed it doesn't work on Ubuntu and MacOS. 
The frontend reports, for example, `"Error loading file: File /tmp/HLTau_Band7_Continuum.fits does not exist."` However, I am then able to use the file browser to navigate to /tmp and successfully open the image, so the file does actually exist and is accessible by CARTA.

Doing it as `carta ../../tmp/HLTau_Band7_Continuum.fits` works.

On MacOS, I noticed that the /tmp folder is actually a symlink to /private/tmp, but doing `carta /private/tmp/HLTau_Band7_Continuum.fits` doesn't work either, and besides, it doesn't work on Ubuntu.

On Ubuntu, moving the CARTA.AppImage into /tmp and then starting it as `./CARTA.AppImage /tmp/HLTau_Band7_Continuum.fits` works. However, doing `carta --folder=/tmp /tmp/HLTau_Band7_Continuum.fits` doesn't work either. So it seems to suggest it somehow depends on the relative location of the CARTA executable and not how `--folder` or `--root` is defined.

It needs further investigation as to what could be wrong.

**Comment by veggiesaurus** [comment]
@ajm-asiaa after the changes to allow just `?file=<path>` URL parameter, can you check if this works?

**Comment by ajm-ska** [comment]
@veggiesaurus Thanks. That allows me to remove some line from Electron's `main.js` file. But I noticed the problem still remained. It turns out the problem was the logic I was using. It was not picking up all possibilities correctly. (This is assuming we are using an alias to the CARTA desktop version from anywhere) Three possibilities are:
1. Using the full absolute path e.g. `carta /tmp/HLTau_Band7_Continuum.fits`
2. Opening a file in the current directory e.g. `carta HLTau_Band7_Continuum.fits`
3. Opening a file in a deeper directory e.g. `carta Images/HLTau_Band7_Continuum.fits`

So after examining it line by line with debug flags on, my previous way only working with possibility 2 and 3. That is because I was using `if ( arg1.includes('/')`. That could occur in both case 1 and 3. But it was only working for case 3 because I was also adding `${process.cwd()` to get the full path, and hence that is why case 1 failed.

Now I have found a better and simpler way using `startsWith('/')` instead:

>         if ( arg1.startsWith('/') ) {
>            console.log("DEBUG: Case 1: String contains a / at beginning so must be an absolute path");
>            mainWindow.loadURL(`file:\/\/${__dirname}\/index.html?file=${arg1}`);
>         } else {
>            console.log("DEBUG: Case 2: Not an absolute path, need to add the current directory");
>            mainWindow.loadURL(`file:\/\/${__dirname}\/index.html?file=${process.cwd()}\/${arg1}`);
>        };

It now works for all three cases. 

So even though it turns out removing the need for `folder` wasn't the cause of this particular issue, it still simplyfies the main.js file quite a bit, so thanks for that!

While you are working on these file nice browser improvements, would it be a good time to address [frontend issue #894](https://github.com/CARTAvis/carta-frontend/issues/894)? (Stopping the file browser appearing when using the URL to directly open a file)
